frontend/src/concepts/pipelines/apiHooks/usePipelineRunById.ts

```diff
@@ -8,8 +8,36 @@ import { usePipelinesAPI } from '~/concepts/pipelines/context';
 import { PipelineRunKFv2, RuntimeStateKF, runtimeStateLabels } from '~/concepts/pipelines/kfTypes';
 import { FAST_POLL_INTERVAL } from '~/utilities/const';
 import { computeRunStatus } from '~/concepts/pipelines/content/utils';
+import { allSettledPromises } from '~/utilities/allSettledPromises';
 
-const usePipelineById = (
+/**
+ * Fetches pipeline runs by their ids, and only returns the successful ones.
+ *
+ * @param pipelineRunIds
+ * @returns all successfully fetched pipeline runs
+ */
+export const usePipelineRunsByIds = (
+  pipelineRunIds: string[],
+): [
+  FetchState<PipelineRunKFv2[]>[0],
+  FetchState<PipelineRunKFv2[]>[1],
+  FetchState<PipelineRunKFv2[]>[3],
+] => {
+  const { api } = usePipelinesAPI();
+
+  const call = React.useCallback<FetchStateCallbackPromise<PipelineRunKFv2[]>>(
+    (opts) =>
+      allSettledPromises(pipelineRunIds.map((id) => api.getPipelineRun(opts, id))).then(
+        ([successful]) => successful.map(({ value }) => value),
+      ),
+    [api, pipelineRunIds],
+  );
+
+  const [data, loaded, , refresh] = useFetchState(call, []);
+  return [data, loaded, refresh];
```

Comment:
Is there a reason we wouldn't return an error if one exists?

Reply:
allSettledPromises wont throw an error, it splits calls into successful responses and error responses. And in our use case we still want the data to be returned despite there being an error on some of the calls (becuase we trim out the ids that returned an error from the url). I went many different directions with this. and becuase of how useFetchState works, having an error means that there cant also be data loaded in.

Reply:
If an error exists, then I would expect no data. So I'm not sure what the issue is there as you're describing how useFetchState works. 

I'm not seeing why we can't also use error responses from allSettledPromises.

Reply:
The function seems tailored then to this 1 use-case but yet has a name that implies it would behave just like any of our other API call hooks.

Reply:
In some fashion the hook needs to give back data despite there being an error, if not then there would be no way to prune the bad runs. I can move this hook into the compare runs directory and rename it to match this usecase.

Reply:
i ended up just pruning it down and putting it in the context provider component becuase no one else would use it